### PR TITLE
Feature/bdkks 15

### PR DESCRIPTION
**Pull request by bmxlugo**
A lot of changes i took from nelmio@67b7b46

### CODE REVIEW

File: Form/Extension/DescriptionFormTypeExtension.php [hunk 1]
```diff
@@ -50,6 +59,6 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
      */
     public function getExtendedType()
     {
-        return 'form';
+        return LegacyFormHelper::getType('Symfony\Component\Form\Extension\Core\Type\FormType');
```
Comment (bmxjama) [inline Form/Extension/DescriptionFormTypeExtension.php:62]: Please change to FormType::class

File: Parser/FormTypeParser.php [hunk 2]
```diff
@@ -52,9 +62,57 @@ class FormTypeParser implements ParserInterface
...
     );
 
-    public function __construct(FormFactoryInterface $formFactory)
+    /**
+     * @var array
+     */
+    protected $extendedMapTypes = array(
+        DataTypes::STRING => array(
+            'text',
+            'Symfony\Component\Form\Extension\Core\Type\TextType',
```
Comment (bmxjama) [inline Parser/FormTypeParser.php:71]: Please change to
use Symfony\Component\Form\Extension\Core\Type;

Type\TextType::class

File: Form/Extension/DescriptionFormTypeExtension.php (hunk 1 above)
Comment (bmxlugo) [inline Form/Extension/DescriptionFormTypeExtension.php:62]: I took it from original repository, i would like not to change to our approach to give a easiest possiblillity to upgrade later.

File: Tests/Fixtures/Form/ImprovedTestType.php [hunk 3]
```diff
@@ -14,35 +25,74 @@ class ImprovedTestType extends AbstractType
...
-            ->add('dt3', 'datetime', array('widget' => 'single_text', 'format' => 'M/d/y H:i:s'))
-            ->add('dt4', 'datetime', array('date_format' => \IntlDateFormatter::MEDIUM))
-            ->add('dt5', 'datetime', array('format' => 'M/d/y H:i:s'))
-            ->add('d1', 'date', array('format' => \IntlDateFormatter::MEDIUM))
-            ->add('d2', 'date', array('format' => 'd-M-y'))
-            ->add('c1', 'choice', array('choices' => array('m' => 'Male', 'f' => 'Female')))
-            ->add('c2', 'choice', array('choices' => array('m' => 'Male', 'f' => 'Female'), 'multiple' => true))
-            ->add('c3', 'choice', array('choices' => array()))
-            ->add('c4', 'choice', array('choice_list' => new SimpleChoiceList(array('foo' => 'bar', 'bazgroup' => array('baz' => 'Buzz')))))
+            ->add('dt1', $datetimeType, array('widget' => 'single_text', 'description' => 'A nice description'))
```
Comment (bmxjama) [inline Tests/Fixtures/Form/ImprovedTestType.php:33]: why array() not [] ?

Comment (bmxlugo) [inline Tests/Fixtures/Form/ImprovedTestType.php:33]: @bmxjama I took it from original repository, i would like not to change to our approach to give a easiest possiblillity to upgrade later.

File: composer.json [hunk 4]
```diff
@@ -15,24 +15,26 @@
         }
     ],
     "require": {
-        "symfony/twig-bundle": "~2.1",
-        "symfony/framework-bundle": "~2.1",
-        "symfony/console": "~2.1",
+        "php": ">=5.3",
+        "symfony/twig-bundle": "~2.3|~3.0",
```
Comment (bmxjama) [inline composer.json]: I think there should be ~2.3||~3.0

File: Util/LegacyFormHelper.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+ */
+
+namespace Nelmio\ApiDocBundle\Util;
+
+/**
+ * Extracted from FOSUserBundle.
+ *
+ * @internal
+ */
+final class LegacyFormHelper
```
Comment (bmxjama) [inline Util/LegacyFormHelper.php:19]: Why you are not using short syntax here, it is not taken from original repository ;)

File: Tests/Parser/FormTypeParserTest.php [hunk 6]
```diff
@@ -500,7 +523,7 @@ public function dataTestParse()
                 ),
             ),
             array(
-                array('class' => 'Nelmio\ApiDocBundle\Tests\Fixtures\Form\DependencyType'),
+                array('class' => 'Nelmio\ApiDocBundle\Tests\Fixtures\Form\DependencyType', 'options' => array(),),
```
Comment (bmxjama) [inline Tests/Parser/FormTypeParserTest.php]: Please change to []

File: Tests/Parser/FormTypeParserTest.php [hunk 7]
```diff
@@ -475,7 +498,7 @@ public function dataTestParse()
                 ),
             ),
             array(
-                array('class' => 'Nelmio\ApiDocBundle\Tests\Fixtures\Form\RequireConstructionType'),
+                array('class' => 'Nelmio\ApiDocBundle\Tests\Fixtures\Form\RequireConstructionType', 'options' => array(),),
```
Comment (bmxjama) [inline Tests/Parser/FormTypeParserTest.php]: Please change to []

File: composer.json (hunk 4 above)
Comment (bmxlugo) [inline composer.json]: @bmxjama  I took it from https://github.com/nelmio/NelmioApiDocBundle/commit/67b7b46627e240cc74c48999ea6a0dafe959e9d5

File: Util/LegacyFormHelper.php (hunk 5 above)
Comment (bmxlugo) [inline Util/LegacyFormHelper.php:19]: It's taken from original!

**Comment by bmxjama** [comment]
:+1:

File: Parser/FormTypeParser.php [hunk 8]
```diff
@@ -253,8 +363,20 @@ private function parseForm($form)
...
+                    } elseif ($choiceList = $config->getOption('choice_list')) {
+                        $choiceListType = $config->getType();
+                        $choiceListName = method_exists($choiceListType, 'getBlockPrefix') ?
+                            $choiceListType->getBlockPrefix() : $choiceListType->getName();
+
+
+                        if (('entity' === $choiceListName && false === $this->entityToChoice)) {
+                            $choices = array();
+                        } else {
+                            // TODO: fixme
```
Comment (bmxdasz) [inline Parser/FormTypeParser.php:375]: This is our `TODO` comment? If yes, we should add some ticket into backlog.

**Comment by bmxdasz** [comment]
File `phpunit.phar` should part of this changes?

File: Parser/FormTypeParser.php (hunk 8 above)
Comment (bmxlugo) [inline Parser/FormTypeParser.php:375]: @bmxdasz i took it from https://github.com/nelmio/NelmioApiDocBundle/blob/master/Parser/FormTypeParser.php

**Comment by bmxlugo** [comment]
I will remove it

**Comment by bmxdasz** [comment]
:+1:

**Pull request merged**